code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -406,42 +406,46 @@
 	togglelock(user)
 
 /mob/living/silicon/robot/attackby(obj/item/W, mob/user, params)
+	if(length(user.progressbars))
+		if(W.tool_behaviour == TOOL_WELDER || istype(W, /obj/item/stack/cable_coil))
+			user.changeNext_move(CLICK_CD_MELEE)
+			to_chat(user, "<span class='notice'>You are already busy!</span>")
+			return
 	if(W.tool_behaviour == TOOL_WELDER && (user.a_intent != INTENT_HARM || user == src))
 		user.changeNext_move(CLICK_CD_MELEE)
 		if (!getBruteLoss())
 			to_chat(user, "<span class='warning'>[src] is already in good condition!</span>")
 			return
-		if (!W.tool_start_check(user, amount=0)) //The welder has 1u of fuel consumed by it's afterattack, so we don't need to worry about taking any away.
-			return
-		if(src == user)
-			to_chat(user, "<span class='notice'>You start fixing yourself.</span>")
-			if(!W.use_tool(src, user, 50))
+		while (getBruteLoss() && W.tool_start_check(user, amount=0))	//Repeatedly attempt to repair dents until done or welder is out of fuel, just like tend wounds. No need to spam click
+			if(src == user)
+				to_chat(user, "<span class='notice'>You start fixing yourself.</span>")
+			if(!W.use_tool(src, user, 60))
 				return
-
-		adjustBruteLoss(-30)
-		updatehealth()
-		add_fingerprint(user)
-		user.visible_message("[user] has fixed some of the dents on [src].", "<span class='notice'>You fix some of the dents on [src].</span>")
+			W.use(1)  //Due to the loop, welding fuel was only being burned passively and not actually consumed each step. This causes it to use 1 every step. 
+			adjustBruteLoss(-15)
+			updatehealth()
+			add_fingerprint(user)
+			user.visible_message("[user] has fixed some of the dents on [src].", "<span class='notice'>You fix some of the dents on [src].</span>")
 		return
 
 	else if(istype(W, /obj/item/stack/cable_coil) && wiresexposed)
 		user.changeNext_move(CLICK_CD_MELEE)
 		var/obj/item/stack/cable_coil/coil = W
-		if (getFireLoss() > 0 || getToxLoss() > 0)
-			if(src == user)
-				to_chat(user, "<span class='notice'>You start fixing yourself.</span>")
-				if(!do_after(user, 50, target = src))
-					return
+		if (!getFireLoss())
+			to_chat(user, "The wires seem fine, there's no need to fix them.")
+			return
+		while(getFireLoss())		//Repeatedly attempt to repair wires until done, just like tend wounds. No need to spam click
+			if(!do_after(user, 60, target = src))
+				return
 			if (coil.use(1))
-				adjustFireLoss(-30)
-				adjustToxLoss(-30)
+				adjustFireLoss(-15)
 				updatehealth()
 				user.visible_message("[user] has fixed some of the burnt wires on [src].", "<span class='notice'>You fix some of the burnt wires on [src].</span>")
 			else
 				to_chat(user, "<span class='warning'>You need more cable to repair [src]!</span>")
-		else
-			to_chat(user, "The wires seem fine, there's no need to fix them.")
-
+				return
+		return
```

Comment:
These returns are pointless atm since everything is in else blocks anyway

Reply:
👍